File: LC/lc48/lc48.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int l = matrix.size();
        for (int x = 0; x < l/2; x++) {
            for (int y = 0; y < l/2+l%2; y++) {
                swap(matrix[x][y], matrix[l-1-y][x]);
                swap(matrix[l-1-y][x], matrix[l-1-x][l-1-y]);
                swap(matrix[l-1-x][l-1-y], matrix[y][l-1-x]);
            }
        }
    }
};

File: LC/lc48/lc48.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        i = n**2//4
        l = n//2 + n%2
        for p in range(i):
            r, c = p//l, p%l
            n1, n2, n3, n4 = matrix[r][c], matrix[c][n-r-1], matrix[n-r-1][n-c-1], matrix[n-c-1][r]
            matrix[r][c], matrix[c][n-r-1], matrix[n-r-1][n-c-1], matrix[n-c-1][r] = n4, n1, n2, n3
        return